plugins/modules/jenkins_credential.py

```diff
@@ -0,0 +1,831 @@
+          - "excluded-hostname:7000"
+          - "excluded-hostname2:7000"
+
+    - name: Add userAndPass credential
+      jenkins_credential:
+        scope: "CUSTOM"
+        id: "userpass-id"
+        type: "userAndPass"
+        jenkinsUser: "admin"
+        token: "token"
+        description: "User and password credential"
+        username: "user1"
+        password: "pass1"
+
+    - name: Add file credential
+      jenkins_credential:
+        id: "file-id"
+        type: "file"
+        jenkinsUser: "admin"
+        token: "token"
+        description: "File credential"
+        filePath: "my-secret.pem"
+
+    - name: Add text credential
+      jenkins_credential:
+        id: "text-id"
+        type: "text"
+        jenkinsUser: "admin"
+        token: "token"
+        description: "Text credential"
+        secret: "mysecrettext"
+
+    - name: Add githubApp credential
+      jenkins_credential:
+        id: "githubapp-id"
+        type: "githubApp"
+        jenkinsUser: "admin"
+        token: "token"
+        description: "GitHub App credential"
+        appID: "12345"
+        filePath: "my-secret.pem"
+        owner: "github_owner"
+
+    - name: Add sshKey credential
+      jenkins_credential:
+        id: "sshkey-id"
+        type: "sshKey"
+        jenkinsUser: "admin"
+        token: "token"
+        description: "SSH key credential"
+        username: "sshuser"
+        filePath: "my-secret.pem"
+        passphrase: "passphrase"
+
+    - name: Add certificate credential (p12)
+      jenkins_credential:
+        id: "certificate-id"
+        type: "certificate"
+        jenkinsUser: "admin"
+        token: "token"
+        description: "Certificate credential"
+        password: "12345678901234"
+        filePath: "certificate.p12"
+
+    - name: Add certificate credential (pem)
+      jenkins_credential:
+        id: "certificate-id-pem"
+        type: "certificate"
+        jenkinsUser: "admin"
+        token: "token"
+        description: "Certificate credential (pem)"
+        filePath: "cert.pem"
+        privateKeyPath: "private.key"
+
+    - name: Delete credential
+      jenkins_credential:
+        id: "credential-id"
+        command: "delete"
+        jenkinsUser: "admin"
+        token: "{{ token }}"
+
+    - name: Delete scope
+      jenkins_credential:
+        id: "scope-id"
+        command: "delete"
+        type: "scope"
+        jenkinsUser: "ruff"
+        token: "{{ token }}"
+
+"""
+# Edit is done the same way as the add command except that the "command" parameter is set to "update".
+# The rest of the parameters are the same as for the add command.(id must be the same as the one used to add the credential)
+
+RETURN = r"""
+changed:
+    description: Whether a change was made.
+    type: bool
+    returned: always
+message:
+    description: Message of the result of the operation.
+    type: str
+    returned: always
+details:
+    description: Incase of errors return more details
```

Comment:
```suggestion
    description: Incase of errors return more details.
```
Generally descriptions (except short_description) should contain complete sentences (when a single string), or multiple paragraphs having complete sentences (when a list of strings).